app.js

```diff
@@ -77,21 +78,23 @@ passport.use(new GoogleStrategy({
 ));
 
 passport.use(new LocalStrategy({
+    passReqToCallBack: true,
     usernameField: 'email',
     passwordField: 'password'
 }, function (email, password, done) {
     if (!email || !password)
         return done({ 'err': { "msg": "Incomplete data", "code": "" } }, null);
```

Comment:
## Query raised on slack 
In case of an Error interacting with our database, we need to invoke `done(err)`. When we cannot find the user or the passwords do not watch, we invoke `done(null, false)`. If everything went fine and we want the user to login we invoke `done(null, user)`.
if you wish to pass additinal info then you can pass it like for example `done(null, false, { 'err': { "msg": "This user does not exist", "code": "" } });`